airbyte-api/src/main/openapi/config.yaml

```diff
@@ -3987,8 +4023,100 @@ components:
           $ref: "#/components/schemas/ConnectionId"
         state:
           $ref: "#/components/schemas/ConnectionStateObject"
+        globalState:
+          $ref: "#/components/schemas/ConnectionStateObject"
+        streamState:
+          type: array
+          $ref: "#/components/schemas/StreamStateObject"
     ConnectionStateObject:
       type: object
+    StreamStateObject:
+      type: object
+      required:
+        - streamKey
+      properties:
+        streamKey:
+          $ref: "#/components/schemas/StreamKey"
+        state:
+          $ref: "#/components/schemas/ConnectionStateObject"
+    ConnectionStateType:
```

Comment:
i think in the new world, we do actually need a add legacy to this enum.

Reply:
done